src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -288,27 +442,86 @@ object JsonProtocol {
           throw new InvalidAnnotationJSONException(s"Expected field 'class' not found! $obj")
         case JObject(fields) => findTypeHints(fields.map(_._2))
         case JArray(arr)     => findTypeHints(arr)
-        case oJValue         => Seq()
+        case _               => Seq()
       })
       .distinct
 
+    // I don't much like this var here, but it has made it much simpler
+    // to maintain backward compatibility with the exception test structure
+    var classNotFoundBuildingLoaded = false
     val classes = findTypeHints(annos, true)
-    val loaded = classes.map(Class.forName(_))
+    val loaded = classes.map { x =>
+      try {
+        Class.forName(x)
+      } catch {
+        case _: java.lang.ClassNotFoundException =>
+          classNotFoundBuildingLoaded = true // tells us which Exception to throw in recovery
+          // Found an annotation we don't recognize, So add UnrecognizedAnnotation to `loaded`
+          Class.forName("firrtl.annotations.UnrecognizedAnnotation")
```

Comment:
Why are we adding `UnrecognizedAnnotation` here? I believe this basically doesn't do anything. Why not just change to a `flatMap` and do `None` here?

Reply:
I found this to be necessary to get things to work. I tried using flatMap several different ways and it just didn't work right. I used several variants of 
```scala
    val loaded = classes.flatMap { x =>
      try {
        Some(Class.forName(x))
      } catch {
        case _: java.lang.ClassNotFoundException =>
          classNotFoundBuildingLoaded = true 
          None
      }
    }
```
 but always got some form of 
```
[error] /Users/chickm/SiFive/code/firrtl/src/main/scala/firrtl/annotations/JsonProtocol.scala:331:33: no type parameters for method flatMap: (f: String => scala.collection.GenTraversableOnce[B])(implicit bf: scala.collection.generic.CanBuildFrom[List[String],B,That])That exist so that it can be applied to arguments (String => Iterable[Class[?0]] forSome { type ?0 })
```

Reply:
Type inference is struggling, it works if you give it a hint:
```scala
    val loaded = classes.flatMap { x =>
      (try {
        Some(Class.forName(x))
      } catch {
        case _: java.lang.ClassNotFoundException =>
          classNotFoundBuildingLoaded = true 
          None
      }): Option[Class[_]]
    }
```
That being said the current code is probably fine, likely marginally slower than flattening them out but I doubt it's even measurable.

Reply:
The hint solved it, thanks